src/hooks/useReferrer.ts

```diff
@@ -15,32 +16,32 @@ export default function useReferrer() {
   const [error, setError] = useState<string>("");
   useEffect(() => {
     setError("");
-    if (provider && r) {
+    if (isDefined(r)) {
       if (ethers.utils.isAddress(r)) {
         setIsResolved(true);
         return setAddress(r);
+      } else if (isDefined(provider)) {
```

Comment:
NIT: If you change this to an `else if` then it would make sense to change line 22 to
```suggestion
        setAddress(r);
      } else if (isDefined(provider)) {
```

Reply:
Yeah, that makes sense! This was old code that I didn't want to update, but the return statement needs to make more sense. https://github.com/across-protocol/frontend-v2/pull/896/commits/e377de57167f67edc32541d3f1f1700de59688ba